apps/prairielearn/src/pages/instructorAssessmentAccessOverrides/instructorAssessmentAccessOverrides.sql

```diff
@@ -0,0 +1,105 @@
+-- BLOCK select_assessment_access_policies
+SELECT
+  format_date_full_compact (aap.created_at, 'America/Chicago') AS created_at,
+  aap.created_by::text AS created_by,
+  aap.credit::text AS credit,
+  format_date_full_compact (aap.end_date, 'America/Chicago') AS end_date,
+  aap.note AS note,
+  format_date_full_compact (aap.start_date, 'America/Chicago') AS start_date,
+  (
+    SELECT
+      name
+    from
+      groups
+    where
+      id = aap.group_id
+  ) as group_name,
+  (
+    SELECT
+      uid
+    FROM
+      users
+    WHERE
+      user_id = aap.user_id
+  ) as student_uid,
+  aap.id as id
+FROM
+  assessment_access_policies AS aap
+WHERE
+  assessment_id = $assessment_id
+ORDER BY
+  aap.created_at;
+
+-- BLOCK select_group_in_assessment
+SELECT
+  groups.id,
+  groups.name,
+  groups.group_config_id
+FROM
+  groups
+  JOIN group_configs ON groups.group_config_id = group_configs.id
+WHERE
+  groups.course_instance_id = $course_instance_id
+  AND groups.name = $group_name
+  AND group_configs.assessment_id = $assessment_id;
+
+-- BLOCK insert_assessment_access_policy
+INSERT INTO
+  assessment_access_policies (
+    assessment_id,
+    created_at,
+    created_by,
+    credit,
+    end_date,
+    group_id,
+    note,
+    start_date,
+    user_id
+  )
+VALUES
+  (
+    $assessment_id,
+    NOW(),
+    $created_by,
+    $credit,
+    $end_date,
+    $group_id,
+    $note,
+    $start_date,
+    $user_id
+  );
+
+-- BLOCK update_assessment_access_policy
+UPDATE assessment_access_policies
+SET
+  created_at = NOW(),
+  credit = $credit,
+  end_date = $end_date,
+  group_id = $group_id,
+  note = $note,
+  start_date = $start_date,
+  user_id = (
+    SELECT
+      user_id
+    FROM
+      users
+    WHERE
+      uid = $student_uid
+  )
```

Comment:
As with `insert_assessment_access_policy`, you need to change this to use a validated `user_id` passed in from JavaScript.

Reply:
Question: 52ad2e9197df6e4f0d1f3b03edb191954f55a44f has the correction of passing the user_id . However, I am worried that I am searching all users, not just those in enrolled in the course. I also think it is redundant to get the user for a group instance. However, I am defining user before params because params is initialized as const, and I am unsure how to initialize user_id without setting it to null.

Reply:
Yeah, you are indeed searching all users. You have this more correct in `add_new_override`, where you're validating with `userIsEnrolledInCourseInstance`. You should do the same here.

Reply:
So would i do something more like this before defining edit_params 
```javascript
const isEnrolled = await userIsEnrolledInCourseInstance({
      uid: req.body.student_uid,
      course_instance_id: res.locals.course_instance.id,
    });
    if (!isEnrolled) {
      throw error.make(400, `User ${req.body.student_uid} is not enrolled in this course instance.`);
    }
    const user = await selectUserByUid(req.body.student_uid);
```

Reply:
^ This was implemented in 741cb98c45c16432439618e239fe97c51f846839

Reply:
Followup comment posted here: https://github.com/PrairieLearn/PrairieLearn/pull/9103#discussion_r1501122880